### PR TITLE
feat: MFLT-11010 Renames de-duplication to grouping in docs

### PR DESCRIPTION
**Pull request by Blisse**
### Summary

Replaces the word "de-duplication" with "grouping" in most situations. 
This is a bit less technical of a word, but much more intuitive to understand.

Cleans up the documentation around MEMFAULT_TRACE_EVENT_*

### CODE REVIEW

File: docs/android/introduction.mdx [hunk 1]
```diff
@@ -18,8 +18,8 @@ Android? [Let us know!](mailto:hello@memfault.com)
    [bug reports](https://developer.android.com/studio/debug/bug-report) for
    large snapshots of diagnostic information.
 2. Kernel oops, tombstones, java exceptions, ANRs, WTFs, SELinux policy
-   violations, and other issues are extracted, grouped, de-duplicated, and
-   displayed in a searchable dashboard. Notifications can let you know when
+   violations, and other issues are extracted, intelligently grouped together,
```
Comment (sjp4) [inline docs/android/introduction.mdx:21]: Is `issues` the right word there? Sunds like a `trace`

File: docs/mcu/introduction.mdx [hunk 2]
```diff
@@ -22,10 +22,10 @@ use or connectivity path in place.
 3. The Memfault cloud reassembles data structures from received packets and
    analyzes them to classify the issue, identify a possible root cause, and
    recover items such as variable and function names from the binary data.
-4. The Memfault cloud automatically de-duplicates similar issues. Notifications
-   are used to let you know when problems are detected. From the Memfault web
-   application, individual issues can be drilled into and overall fleet health
-   can be examined via dashboard views.
+4. The Memfault cloud automatically groups similar issues together.
```
Comment (sjp4) [inline docs/mcu/introduction.mdx:25]: Same comment: Memfault doesn't group issues; it groups traces into issues?

Comment (szshen1) [reply docs/mcu/introduction.mdx:25]: +1, we group traces into issues (the DB table is actually called `trace_groups`).

File: docs/android/introduction.mdx (hunk 1 above)
Comment (Blisse) [reply docs/android/introduction.mdx:21]: mmm not capital I issue, though we can be clearer here

File: docs/mcu/trace-events.mdx [hunk 3]
```diff
@@ -106,26 +106,34 @@ For details on the trace event macros, see the
 
 ### Trace Event Variants
 
-As shown above, there are a few different types of Trace Events available.
+As shown above, there are a 3 different types of Trace Events available.
```
Comment (noahp) [inline docs/mcu/trace-events.mdx:109]:
```suggestion
As shown above, there are 3 different types of Trace Events available.
```

File: docs/mcu/trace-events.mdx [hunk 4]
```diff
@@ -106,26 +106,34 @@ For details on the trace event macros, see the
...
+- `MEMFAULT_TRACE_EVENT` events with the same reason and backtrace will be
+  grouped together.
+- `MEMFAULT_TRACE_EVENT_WITH_STATUS` events with the same Reason, the same
+  Status code, and the same backtrace are grouped together.
+- `MEMFAULT_TRACE_EVENT_WITH_LOG` with the same Reason and backtrace will be
+  grouped together. The log is **not** used for grouping. `MEMFAULT_TRACE_EVENT`
+  and `MEMFAULT_TRACE_EVENT_WITH_LOG` events may be grouped together if their
+  reason and backtrace are the same, regardless of the log.
+
+It is recommended to use `MEMFAULT_TRACE_EVENT_WITH_STATUS()` to explicitly
```
Comment (noahp) [inline docs/mcu/trace-events.mdx:132]: this is great, thank you!

**Review by noahp — APPROVED** [review]

**Pull request merged**